specs/BotBuilderOpenSourcePlans.md

```diff
@@ -0,0 +1,85 @@
+# BotBuilder Open Source Engagement
+
+## Goals
+The BotBuilder SDK is an open-source project on GitHub that aspires to:
+
+* Build a thriving community of developers who build bots. 
+* Align 1st and 3rd party developers
+* Provide the tools and processes needed to support our development community. 
+* Leverage existing best-practices wherever they are found. 
+* Keep all platform implementations of the Botbuilder in Sync.
+
+## Non-Goals:
+* TBD
+
+# Exemplar Projets
+From an open-source community perspective, the BotBuilder is using several projects as a North Star. 
+1. [Typescript](https://github.com/Microsoft/TypeScript)
+2. [VS Code](https://github.com/Microsoft/vscode)
+3. [.Net Core Fx](https://github.com/dotnet/corefx)
+4. [.Net Core CLR](https://github.com/dotnet/coreclr)
+5. [ASP.Net](https://github.com/aspnet/Home)
+6. [TensorFlow](https://github.com/tensorflow/tensorflow)
+7. [Bootstrap](https://github.com/twbs/bootstrap)
+
+# Builds
+Many popular project have daily builds. This includes Typescript, [TensorFlow](https://pypi.org/project/tf-nightly/), [VS Code](https://code.visualstudio.com/insiders/)and [.Net Core FX](https://dotnet.myget.org/gallery/dotnet-core). 
+
+Nightly builds enabled features, bug fixes, and external pull requsts to more rapidly reach the interested audiance. The BotBuilder SDK enables nightly builds via MyGet. To facilitate nightly builds, the team has enabled [botbuilder.myget.org](https://botbuilder.myget.org) and has created a daily build gallery for each of:
+1. [C# SDK v4](https://botbuilder.myget.org/gallery/botbuilder-v4-dotnet-daily)
+2. [Node SDK v4](https://botbuilder.myget.org/gallery/botbuilder-v4-js-daily)
+3. [Java SDK v4](https://botbuilder.myget.org/gallery/botbuilder-v4-java-daily)
+
+Following the Python Best practices as defined by TensorFlow, the daily builds for the Python SDK will be here:
+[https://pypi.org/project/botbuilder-nightly/](https://pypi.org/project/botbuilder-nightly/)
+
+# External Pull Requests
+
+We love external pull requests. For submitted PR's we require:
+1. Follow the contribution guidelines
+2. Signed the relevant CLA
+3. Follow coding standards for relevant platform
+4. Unit Tests all pass. 
+
+The BotBuilder team is following the [.Net Core FX/CLR contribution guidelines](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/contributing.md). 
```

Comment:
@cleemullins are we going to ask folks to have PR for each lang or we are open to taking PR in a single language? Do to long-term maintenance code, we should include a clear guidance on how the team evaluates PR.

Reply:
I second @yochay here.  If something is a bug fix specific to only one language, it makes sense to only make the change in one language.  What about something like an enhancement though?

Reply:
No customer will have any interest in porting their changes to 4 (or more) platforms. That's our job. I don't see how we can ask that of them.